src/main/webapp/register.js

```diff
@@ -14,9 +14,23 @@
 
 function onRegisterLoad() {
   supplyLogoutLink();
+  checkRegisteredBefore();
   autofillEmail().then(prepareFormValidation());
 }
 
+// If they have registered before, redirect them to profile page.
```

Comment:
This seems to check whether the user is currently logged in, not whether they have previously registered? If so, please change the comment and function name.

Reply:
No it checks whether they have previously registered. Added more docu.